set.mm

```diff
@@ -142620,6 +142644,16 @@ notation is used by Donald Knuth for iterated exponentiation (_Science_
       ZBCLZHABMNOBPQCMNOCPQUBUCRDFEGBCSTUA $.
   $}
 
+  $( TODO - challenge to shorten this proof! $)
+  $( A natural number with square one is equal to one.  (Contributed by Thierry
+     Arnoux, 2-Feb-2020.) $)
+  nn0sqeq1 $p |- ( ( N e. NN0 /\ ( N ^ 2 ) = 1 ) -> N = 1 ) $=
+    ( cn0 wcel c2 cexp co c1 wceq wne clt wbr wo nn0re 1red lttri2d wa nn0lt10b
+    cc0 biimpa a1i sq0id 0ne1 eqnetrd 0le1 nn0ge0 lt2sqd syl5eqbrr gtned jaodan
+    sq1 cle ex sylbid necon4d imp ) ABCZADEFZGHAGHUPAGUQGUPAGIAGJKZGAJKZLZUQGIZ
+    UPAGAMZUPNZOUPUTVAUPURVAUSUPURPZUQRGVDAUPURARHAQSUARGIVDUBTUCUPUSPZGUQVENVE
+    GGDEFZUQJUJUPUSVFUQJKUPGAVCVBRGUKKUPUDTAUEUFSUGUHUIULUMUNUO $.
+
```

Comment:
The shorter proof is now in place.

Reply:
Indeed now the proof depends on the definition of square root, which is maybe why I had a longer proof in the first place: I wanted to stay in the realm of natural numbers.

However since nothing else is moving, I think this proof is fine.